builtInFunctions/esdtMetaDataRecreate.go

```diff
@@ -0,0 +1,189 @@
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+) (*esdtMetaDataRecreate, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(storageHandler) {
+		return nil, ErrNilESDTNFTStorageHandler
+	}
+	if check.IfNil(rolesHandler) {
+		return nil, ErrNilRolesHandler
+	}
+
+	e := &esdtMetaDataRecreate{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		storageHandler:        storageHandler,
+		rolesHandler:          rolesHandler,
+		enableEpochsHandler:   enableEpochsHandler,
+		funcGasCost:           funcGasCost,
+		gasConfig:             gasConfig,
+		mutExecution:          sync.RWMutex{},
+	}
+
+	e.baseActiveHandler.activeHandler = enableEpochsHandler.IsDynamicESDTEnabled
+
+	return e, nil
+}
+
+func checkArguments(vmInput *vmcommon.ContractCallInput, acntSnd vmcommon.UserAccountHandler, handler baseActiveHandler) error {
+	if vmInput == nil {
+		return ErrNilVmInput
+	}
+	if vmInput.CallValue == nil {
+		return ErrNilValue
+	}
+	if vmInput.CallValue.Cmp(zero) != 0 {
+		return ErrBuiltInFunctionCalledWithValue
+	}
+	if !bytes.Equal(vmInput.CallerAddr, vmInput.RecipientAddr) {
+		return ErrInvalidRcvAddr
+	}
+	if check.IfNil(acntSnd) {
+		return ErrNilUserAccount
+	}
+	if !handler.IsActive() {
+		return ErrBuiltInFunctionIsNotActive
+	}
+
+	return nil
+}
+
+func getEsdtDataAndCheckType(
+	vmInput *vmcommon.ContractCallInput,
+	acntSnd vmcommon.UserAccountHandler,
+	storageHandler vmcommon.ESDTNFTStorageHandler,
+) (*esdt.ESDigitalToken, []byte, uint64, error) {
+	esdtTokenKey := append([]byte(baseESDTKeyPrefix), vmInput.Arguments[tokenIDIndex]...)
+	nonce := big.NewInt(0).SetBytes(vmInput.Arguments[nonceIndex]).Uint64()
+	esdtData, err := storageHandler.GetESDTNFTTokenOnSender(acntSnd, esdtTokenKey, nonce)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	if !core.IsDynamicESDT(esdtData.Type) {
+		return nil, nil, 0, ErrOperationNotPermitted
+	}
+
+	return esdtData, esdtTokenKey, nonce, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtMetaDataRecreate) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	err := checkArguments(vmInput, acntSnd, e.baseActiveHandler)
+	if err != nil {
+		return nil, err
+	}
+	if len(vmInput.Arguments) < 7 {
+		return nil, ErrInvalidNumberOfArguments
+	}
+
+	err = e.rolesHandler.CheckAllowedToExecute(acntSnd, vmInput.Arguments[tokenIDIndex], []byte(core.ESDTMetaDataRecreate))
+	if err != nil {
+		return nil, err
+	}
+
+	totalLength := uint64(0)
+	for _, arg := range vmInput.Arguments {
+		totalLength += uint64(len(arg))
+	}
+
+	e.mutExecution.RLock()
+	gasToUse := totalLength*e.gasConfig.StorePerByte + e.funcGasCost
```

Comment:
take store per byte for difference only.

Reply:
Done.